--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -2213,13 +2213,16 @@
 	int i;
 	int val;
 
-	for (i = 0; i < 32; i++) {
+	usleep_range(9500, 10000);
+
+	for (i = 0; i < 250; i++) {
 		val = __phy_read(phydev, MII_VCT5_CTRL);
 		if (val < 0)
 			return val;
 
 		if (val & MII_VCT5_CTRL_COMPLETE)
 			return 0;
+		usleep_range(3000, 3500);
 	}
 
 	phydev_err(phydev, "Timeout while waiting for cable test to finish\n");
